src/Build/Graph/GraphBuilder.cs

```diff
@@ -280,8 +281,22 @@ private static void AddEdgesFromSolution(IReadOnlyDictionary<ConfigurationMetada
 
             SolutionConfigurationInSolution currentSolutionConfiguration = SelectSolutionConfiguration(solution, solutionEntryPoint.GlobalProperties);
 
+            // Mimic behavior of SolutionProjectGenerator
             string solutionConfigurationXml = SolutionProjectGenerator.GetSolutionConfiguration(solution, currentSolutionConfiguration);
             solutionGlobalPropertiesBuilder["CurrentSolutionConfigurationContents"] = solutionConfigurationXml;
+            solutionGlobalPropertiesBuilder["BuildingSolutionFile"] = "true";
+
+            string solutionDirectoryName = solution.SolutionFileDirectory;
+            if (!solutionDirectoryName.EndsWith(Path.DirectorySeparatorChar.ToString(), StringComparison.Ordinal))
```

Comment:
I'm tempted to recommend EnsureTrailingSlash, but this is how it does it in SPG, so I can't really complain.